### PR TITLE
Add limit and context cancellation handling to current tables. Closes #62

### LINKED ISSUE #62: Add limit and context cancellation handling to current tables
**Is your feature request related to a problem? Please describe.**
When listing results from various tables, too many results are returned or iterated.

**Describe the solution you'd like**
All tables that support limiting results should:
- Set the default page size to the max allowed
- If a `limit` is passed in that's lower than the default, set the page size to `limit` to reduce returned results, e.g., https://github.com/turbot/steampipe-plugin-okta/blob/main/okta/table_okta_application.go#L71-L82
- Handle context cancellations when iterating through the results (in the initial call, and when paging), e.g., https://github.com/turbot/steampipe-plugin-okta/blob/main/okta/table_okta_application.go#L110-L113, https://github.com/turbot/steampipe-plugin-okta/blob/main/okta/table_okta_application.go#L127-L130

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
Tables updated:
- [ ] okta_app_assigned_group
- [ ] okta_app_assigned_user
- [ ] okta_auth_server
- [ ] okta_factor
- [ ] okta_group
- [ ] okta_idp_discovery_policy
- [ ] okta_network_zone
- [ ] okta_password_policy
- [ ] okta_signon_policy
- [ ] okta_trusted_origin
- [ ] okta_user
- [ ] okta_user_type

# Example query results
[output.txt](https://github.com/turbot/steampipe-plugin-okta/files/7949820/output.txt)

### CODE REVIEW

File: okta/table_okta_trusted_origin.go [hunk 1]
```diff
@@ -53,31 +53,55 @@ func listOktaTrustedOrigins(ctx context.Context, d *plugin.QueryData, _ *plugin.
...
-		logger.Error("listOktaTrustedOrigins", "connect", err)
+		logger.Error("listOktaTrustedOrigins", "connect_error", err)
 		return nil, err
 	}
 
-	input := query.Params{}
+	// Maximum limit isn't mentioned in the documentation
+	// Default maximum limit is set as 1000
+	input := query.Params{
+		Limit: 1000,
```
Comment (cbruno10) [inline okta/table_okta_trusted_origin.go:63]: What's the reasoning for picking 1000 as the default?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@c0d3r-arnab Please see comments, thanks!

File: docs/tables/okta_app_assigned_group.md [hunk 2]
```diff
@@ -42,8 +42,8 @@ select
   grp.name as group_name,
   grp.description as group_description,
   jsonb_pretty(grp.group_members) as group_members
-from
-  okta_application app
-left join okta_app_assigned_group ag on app.id = ag.app_id
-left join okta_group grp on ag.id = grp.id;
+from
```
[the remark this answers is not in the document]
Comment (c0d3r-arnab) [reply docs/tables/okta_app_assigned_group.md]: I missed it. I have removed it now.

File: okta/table_okta_trusted_origin.go (hunk 1 above)
Comment (c0d3r-arnab) [reply okta/table_okta_trusted_origin.go:63]: In the other plugins, it was decided to set 1000 as the maximum limit when it is not mentioned in the documentation. Then we will test the query manually and reduce the number if the API doesn't support it. I have followed the same pattern in this case.

Comment (cbruno10) [reply okta/table_okta_trusted_origin.go:63]: It's good to look at other plugins, but we should probably look at Okta's other limits, not other plugins, to base a default off of. What are most defaults for other Okta APIs? 200?

Comment (c0d3r-arnab) [reply okta/table_okta_trusted_origin.go:63]: In most of the cases that I have come across so far, it's 200. I have 10000 and 500 in 2 tables.

Comment (cbruno10) [reply okta/table_okta_trusted_origin.go:63]: Let's go with 200 then, it feels like a safe bet based on other APIs - if we come across different information later on, we can update the default.

Comment (c0d3r-arnab) [reply okta/table_okta_trusted_origin.go]: I have updated the default limits of both `okta_network_zone` and `okta_trusted_origin` to 200.

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**